### ISSUE TITLE
Acciqure and release "exclusive" checkout without undoing it first.

### ISSUE THREAD

**Issue by kitor** [NationalSecurityAgency/ghidra#5430]
### Is your feature request related to a problem? Please describe.
I work in a project where from time to time we need to get exclusive checkout to do stuff like updating memory map, or just upgrading file to a newer ghidra version. The workflow is usually like this:

1. Get all the other people to undo their checkouts or kick other people
2. Check in own changes if any
3. Undo own checkout
4. Checkout with exclusive flag checked _(= network transfer / time wasted )_
5. Do the work
6. Check in the work. 
7. Undo checkout (or unset "keep file checked out" in step 5) _(= network transfer / time wasted )_
8. Checkout again + inform others file can be checked out again

This add IMO unnecessary steps + in case of working with huge projects on remote servers it adds a lot of network transfer and time wasted on those.

### Describe the solution you'd like
Allow user to do two things, ideally from right-click menu in active project view (next to all the check in/out related functions)

#### Acquire "exclusive" flag on existing checkout 
If this is the only existing checkout on the server, allow user to select option to convert it to exclusive.

#### Release "exclusive" flag on existing checkout
If user has exclusive checkout, allow user to select option to convert it to non-exclusive.

I think in case of releasing exclusive flag all user changes would need to be either checked-in or dropped first. 

If implemented, this could be also a part of "check in" GUI, where "keep the file checked out" option lies for now. 

### Describe alternatives you've considered
I see no other alternatives than keeping the existing way of work.

### Additional context
If both options are implemented, workflow would simplify to

1. Get all the other people to undo their checkouts or kick other people
3. Request "exclusive" flag ( no wait / extra transfers ). No need to check-in own changes yet.
4. Do the work
5. Check in the work
6. Release "exclusive" flag ( no wait / extra transfers )
7. Inform others that file can be checked out again.

**Comment by coon42** [comment]
Agreed. Is there a technical reason why a new complete checkout is required when switching between exclusive and non exclusive?

**Comment by ghidra1** [comment]
This is related to #5368 which is alread in our queue

**Comment by ghidra1** [comment]
> Agreed. Is there a technical reason why a new complete checkout is required when switching between exclusive and non exclusive?

The only limitation is the server interface API which does not support this operation at present.  This change will require a change to the server interface which will likely break compatibility with between client/server versions.
